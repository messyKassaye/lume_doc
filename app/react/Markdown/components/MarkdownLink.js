import { Link } from 'react-router';
import PropTypes from 'prop-types';
import React from 'react';

const MarkdownLink = ({ url, classname, children }) => {
  console.warn('MarkdownLink is deprecated, use <Link to="url"></Link> instead');
  return (
    <Link to={url} className={classname} href={url}>
      {children}
    </Link>
  );
};

MarkdownLink.defaultProps = {
  children: '',
  classname: '',
};

MarkdownLink.propTypes = {
  url: PropTypes.string.isRequired,
  classname: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.string,
  ]),
};

export default MarkdownLink;
