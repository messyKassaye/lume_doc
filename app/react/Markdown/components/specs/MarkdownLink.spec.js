import React from 'react';
import { shallow } from 'enzyme';
import MarkdownLink from '../MarkdownLink.js';

describe('Link', () => {
  beforeEach(() => {
    spyOn(console, 'warn');
  });
  it('should render a react-router Link', () => {
    const component = shallow(<MarkdownLink url="url">label</MarkdownLink>);
    expect(component).toMatchSnapshot();
  });

  it('should allow to set a custom class to the component', () => {
    const component = shallow(
      <MarkdownLink url="url" className="custom class">
        label
      </MarkdownLink>
    );
    expect(component).toMatchSnapshot();
  });

  it('should allow nesting children inside', () => {
    const component = shallow(
      <MarkdownLink url="url">
        <div>Extra content</div>
        <span>Label</span>
      </MarkdownLink>
    );
    expect(component).toMatchSnapshot();
  });
});
