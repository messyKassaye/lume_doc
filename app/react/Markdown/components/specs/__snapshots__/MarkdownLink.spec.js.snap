// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should allow nesting children inside 1`] = `
<Link
  className=""
  href="url"
  onlyActiveOnIndex={false}
  style={Object {}}
  to="url"
>
  <div>
    Extra content
  </div>
  <span>
    Label
  </span>
</Link>
`;

exports[`Link should allow to set a custom class to the component 1`] = `
<Link
  className=""
  href="url"
  onlyActiveOnIndex={false}
  style={Object {}}
  to="url"
>
  label
</Link>
`;

exports[`Link should render a react-router Link 1`] = `
<Link
  className=""
  href="url"
  onlyActiveOnIndex={false}
  style={Object {}}
  to="url"
>
  label
</Link>
`;
